### ISSUE TITLE
[expect-expect] Option to enforce the last statement of a test to be an assertion

### ISSUE THREAD

**Issue by CreativeTechGuy** [jest-community/eslint-plugin-jest#1338]
The expect-expect rule works great for identifying a test that doesn't have any assertions, but it doesn't work for identifying halfway-complete tests. Often more complex test cases (such as JSDOM tests) will have multiple assertions, usually before something has occurred to assert the base state is correct, then perform an action, and then assert that the values have changed as expected. I have found a few times where I'll be writing a test and get distracted halfway through and forget to finish the second half of the test. Code coverage looks good since all of the actions were performed, but the test was missing the critical assertions at the end.

In almost every test, there should be an assertion as the last statement. The only case where I can think of that not being true is when the assertions are inside of callback function, but that can be avoided by putting `expect.assertions(number)` as the last line.

Since it seems like a generally useful option which aligns with the spirit of this rule, I wanted to suggest this as a new feature.

**Comment by danielnixon** [comment]
> Code coverage looks good since all of the actions were performed, but the test was missing the critical assertions at the end.

Unhelpful drive-by suggestion: mutation testing can help on this front, e.g. https://stryker-mutator.io/

**Comment by SimenB** [comment]
@G-Rath thoughts on this? I think it sounds like a good idea - not sure if it should be an option on `expect-expect` or a separate rule

**Comment by G-Rath** [comment]
@SimenB yeah it makes sense to me, and probably as an option to `expect-expect` 🤔

**Comment by G-Rath** [comment]
@CreativeTechGuy fyi I've opened #1742 for this
